document-portal/xdp-main.c

```diff
@@ -457,6 +457,11 @@ validate_fd (int fd,
 
       strncpy (path_buffer, app_path, PATH_MAX);
     }
+  else if (g_str_has_prefix (path_buffer, "/newroot/"))
```

Comment:
This should only be in effect if app_id != NULL && *app_id != 0 (i.e. in a sandbox).